lib/middleman-hashicorp/redcarpet.rb

```diff
@@ -69,10 +69,15 @@ def block_html(raw)
 
     if md = raw.match(/\<(.+?)\>(.*)\<(\/.+?)\>/m)
       open_tag, content, close_tag = md.captures
-      "<#{open_tag}>\n#{recursive_render(content)}<#{close_tag}>"
-    else
-      raw
+      rendered_content = recursive_render(content)
+      # If the recursive render did more than simply wrap
+      # the content within a paragraph, assume this is valid
+      # markdown content and return the result
+      if rendered_content != paragraph(content)
+        "<#{open_tag}>\n#{rendered_content}<#{close_tag}>"
+      end
```

Comment:
Sorry I might just not understand how this works very well, but is it the intended course of action that if this block returns false, nothing renders at all?